File: src/Calculator/Controls/RadixButton.cpp
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

#include "pch.h"
#include "RadixButton.h"
#include "CalcViewModel/Common/LocalizationSettings.h"

using namespace CalculatorApp;
using namespace CalculatorApp::Common;
using namespace CalculatorApp::Controls;
using namespace Platform;
using namespace std;
using namespace Windows::Foundation::Collections;
using namespace Windows::System;
using namespace Windows::UI::Xaml;
using namespace Windows::UI::Xaml::Input;
using namespace Windows::UI::Xaml::Data;

String ^ RadixButton::GetRawDisplayValue()
{
    String ^ radixContent = Content->ToString();
    return LocalizationSettings::GetInstance().RemoveGroupSeparators(radixContent);
}

File: src/Calculator/Controls/RadixButton.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using CalculatorApp.ViewModel.Common;
using Windows.UI.Xaml;

namespace CalculatorApp
{
    namespace Controls
    {
        public sealed class RadixButton : Windows.UI.Xaml.Controls.RadioButton
        {
            public RadixButton()
            { }

            internal string GetRawDisplayValue()
            {
                string radixContent = Content?.ToString();
                return LocalizationSettings.GetInstance().RemoveGroupSeparators(radixContent);
            }

            public string Label
            {
                get => (string)GetValue(LabelProperty);
                set => SetValue(LabelProperty, value);
            }

            public static readonly DependencyProperty LabelProperty =
                DependencyProperty.Register(nameof(Label), typeof(string), typeof(RadixButton), new PropertyMetadata(default(string)));
        }
    }
}
